rust/main/submitter/src/payload_dispatcher/dispatcher.rs

```diff
@@ -42,12 +57,89 @@ impl PayloadDispatcher {
         })
     }
 
-    pub fn spawn(self) -> Instrumented<JoinHandle<()>> {
-        // TODO: here
-        // create the submit queue and channels for the Dispatcher stages
+    pub async fn spawn(self) -> JoinHandle<()> {
+        let mut tasks = vec![];
+        let building_stage_queue: BuildingStageQueue = Arc::new(Mutex::new(VecDeque::new()));
+        let (inclusion_stage_sender, inclusion_stage_receiver) =
+            tokio::sync::mpsc::channel::<Transaction>(SUBMITTER_CHANNEL_SIZE);
+        let (finality_stage_sender, finality_stage_receiver) =
+            tokio::sync::mpsc::channel::<Transaction>(SUBMITTER_CHANNEL_SIZE);
+
+        let building_stage = BuildingStage::new(
+            building_stage_queue.clone(),
+            inclusion_stage_sender.clone(),
+            self.inner.clone(),
+        );
+        let building_task = tokio::spawn(
```

Comment:
It would be better to use `tokio::task::Builder::new().name("<name>")` so that each task has a name and we can identify it in Tokio Console.